pkg/apis/clusteroperator/validation/clusterversion.go

```diff
@@ -69,6 +69,18 @@ func ValidateClusterVersionSpec(spec *clusteroperator.ClusterVersionSpec, fldPat
 		allErrs = append(allErrs, field.Required(vmImagesPath, "must define VM images"))
 	}
 
+	deploymentTypePath := fldPath.Child("deploymentType")
+	if spec.DeploymentType == "" {
+		allErrs = append(allErrs, field.Required(deploymentTypePath, "must define deployment type"))
+	} else if spec.DeploymentType != clusteroperator.ClusterDeploymentTypeOrigin && spec.DeploymentType != clusteroperator.ClusterDeploymentTypeEnterprise {
+		allErrs = append(allErrs, field.Invalid(deploymentTypePath, spec.DeploymentType, "unknown deployment type"))
```

Comment:
`field.NotSupported` is a better choice than `field.Invalid` for enumerated fields. See [here](https://github.com/openshift/cluster-operator/blob/551c8e210cb9349d6f709aa303909bc665e09280/pkg/apis/clusteroperator/validation/validation.go#L48) for an example.

Reply:
Fixed.